src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -1449,13 +1494,33 @@ private bool OpenJsonFile(WorkspaceInfo workspaceInfo, string fileContents, out
                 false,
                 CustomNodeManager);
 
-            workspace.FileName = workspaceInfo.FileName;
+            workspace.FileName = filePath;
+            workspace.ScaleFactor = dynamoPreferences.ScaleFactor;
```

Comment:
lets harden this as described above.  - a potential solution to keep this code free of null checks, is that if the deserialization of dynamo block fails, fill it up with default values?

Reply:
I have added code to set default preferences in the case that the Dynamo preferences block is not present.